src/GerdaFitter.cc

```diff
@@ -540,6 +485,126 @@ GerdaFitter::GerdaFitter(json outconfig) : config(outconfig) {
             this->DumpData();
         }
     }
+
+    /*
+     * define observables
+     */
+
+    if (config.contains("observables")) {
+        BCLog::OutDebug("Parsing 'observables' section");
+        for (auto& el : config["observables"].items()) {
+            // sanity checks
+            auto _expr = el.value()["TFormula"].get<std::string>();
+            TFormula _tformula(el.key().c_str(), _expr.c_str());
+            // the following check will always fail with ROOT < 6.12/04
+            // The following fixes are essential for this code to work:
+            // https://root.cern.ch/doc/v612/release-notes.html#histogram-libraries
+            if (!_tformula.IsValid() or _tformula.GetNpar() < 1 or _tformula.GetNdim() > 0) {
+                throw std::runtime_error("invalid observables TFormula given");
+            }
+            // loop over number of parameters in formula
+            for (int p = 0; p < _tformula.GetNpar(); ++p) {
+                std::string parname = _tformula.GetParName(p);
+                bool exists = false;
+                // look for BAT internal parameter index
+                for (unsigned int idx = 0; idx < this->GetNParameters(); ++idx) {
+                    if (this->GetParameters().At(idx).GetName() == parname) {
+                        exists = true;
+                        // we do this here because we'd rather save BAT's internal parameter index
+                        // to access it later in GerdaFitter::CalculateObservables
+                        _tformula.SetParName(p, std::to_string(idx).c_str());
+                        break;
+                    }
+                }
+                // throw exception if parameter is undefined
+                if (!exists) throw std::runtime_error(
+                    "fit parameter '" + parname + "' not found, is it defined in \"parameters\"?"
+                );
+            }
+
+            // if requested substitute all parameters with :
+            // <integral-range> X <parameter> in formula
+            if (el.value().contains("scale-with-pdf-integral")) {
```

Comment:
`multiply-by-pdf-integral` would perhaps be more clear, `scale` is more generic

Reply:
... ne abbiamo parlato e ho messo i nomi che avevamo deciso, se non ti piace cambialo